model-api-gen/src/main/kotlin/org/modelix/metamodel/generator/NameConfigBasedGenerator.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.metamodel.generator
+
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.CodeBlock
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.TypeName
+import com.squareup.kotlinpoet.asClassName
+import com.squareup.kotlinpoet.asTypeName
+import org.modelix.metamodel.GeneratedChildListLink
+import org.modelix.metamodel.GeneratedMandatorySingleChildLink
+import org.modelix.metamodel.GeneratedReferenceLink
+import org.modelix.metamodel.GeneratedSingleChildLink
+import org.modelix.model.data.EnumPropertyType
+import org.modelix.model.data.Primitive
+import org.modelix.model.data.PrimitivePropertyType
+
+internal abstract class NameConfigBasedGenerator(open val nameConfig: NameConfig) {
```

Comment:
Distributing them would cause a lot of redundancy, as most of these functions are used by multiple subgenerators. The point of accumulating these was to have a single source of truth for their implementations.